keps/sig-scheduling/1819-scheduler-extender/README.md

```diff
@@ -0,0 +1,306 @@
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Kubernetes scheduler schedules based on resources managed by Kubernetes.
+Scheduling based on opaque resource counting helps extending this further. But
+when there is a need for contextual scheduling for resources managed outside of
+kubernetes, there is no mechanism to do it today.
+
+The proposal is to make kubernetes scheduler extensible by adding the capability
+to make http calls out to another endpoint to help achieve this functionality.
+
+**NOTE**: This KEP was done in retrospect and it was based on the original
+extender proposal.
+
+## Motivation
+
+There are three ways to add new scheduling rules to Kubernetes:
+
+- Update existing or add new scheduler plugins and recompiling
+- Implementing your own scheduler process that runs instead of, or alongside of, the standard Kubernetes scheduler
+- Implementing a "scheduler extender" process that the standard Kubernetes scheduler calls out to as a final pass when making scheduling decisions
+
+The first and the second approach have a high bar for beginners. Scheduler
+plugins must be written in Go and are compiled with Kubernetes scheduler code.
+Implementing a new scheduler requires reinventing the wheel and can't work with
+the default scheduler.
```

Comment:
"can't work with the default scheduler" is misleading, if the custom scheduler used a different name, then they can work together. I would say: implementing a new scheduler from scratch to replace the default one requires a lot of effort if supporting all features is required (e.g., spreading, affinity, taints etc.)

Reply:
SGTM, updated
thanks for the suggestion!